lib/graphql/schema/default_type_error.rb

```diff
@@ -0,0 +1,19 @@
+module GraphQL
+  class Schema
+    module DefaultTypeError
+      def self.call(value, field, parent_type, query_ctx)
+        if field.type.kind.non_null?
```

Comment:
The parameters to this hook don't actually give you the information about the type that had the error in the case of wrapping types.  For example, the field may be of type `[String!]` in which case the field itself is nullable and drop to the else branch where it assumes it is an unresolved type and try to resolve the `nil` type.

E.g. this script reproduces the bug

```ruby
require 'graphql'

QueryType = GraphQL::ObjectType.define do
  name "Query"

  field :invalid_null_list, types[!types.String] do
    resolve proc { [nil] }
  end
end

Schema = GraphQL::Schema.define do
  query QueryType
end

query = GraphQL::Query.new(Schema, '{ invalid_null_list }')
puts query.result
```

which results in the output

```
/Users/dylansmith/src/graphql-ruby/lib/graphql/schema.rb:218:in `resolve_type': Can't determine GraphQL type for: nil, define `resolve_type (obj, ctx) -> { ... }` inside `Schema.define`. (NotImplementedError)
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/query.rb:198:in `resolve_type'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/schema/default_type_error.rb:12:in `call'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/schema.rb:284:in `type_error'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:129:in `resolve_value'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:156:in `block in resolve_value'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:148:in `each'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:148:in `each_with_index'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:148:in `each'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:148:in `map'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:148:in `resolve_value'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:116:in `continue_resolve_field'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:87:in `resolve_field'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:30:in `block in resolve_selection'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/internal_representation/selection.rb:40:in `block in each_selection'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/internal_representation/selection.rb:37:in `each'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/internal_representation/selection.rb:37:in `each_selection'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:28:in `resolve_selection'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/execution/execute.rb:8:in `execute'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/query/executor.rb:33:in `execute'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/query/executor.rb:15:in `result'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/query.rb:272:in `get_result'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/query.rb:256:in `call'
	from /Users/dylansmith/src/graphql-ruby/lib/graphql/query.rb:102:in `result'
	from tmp/test.rb:16:in `<main>'
```


It would be much nicer to just get the exception object (e.g. GraphQL::InvalidNullError or GraphQL::UnresolvedTypeError) which could inherit from a common base class like GraphQL::TypeError and expose the relevant values that you are currently passing into this lambda.